### PR TITLE
JITMs: Remove per-site cache and per-page distinction for sidebar_notice 

### LINKED ISSUE #46809: JITMs: Remove per-site cache and per-page distinction for sidebar_notice
## What are JITMs:

JITM stands for "Just In Time Messages" and is a flexible system that communicates with the backend to show messages like "Free Domain with a plan [Upgrade]". It works on a per site, per page basis.  See more in #46677

## Desired Changes

For simplicity:
* Remove the per-site cache added in #46677
* Move all `calypso:\w+:sidebar_notice` messagePaths to use the same messagePath

## Reasoning
See the last few comments in #46677

### PR DESCRIPTION
**Pull request by mreishus**
fixes #46809

#### Changes proposed in this Pull Request

* Remove the per-site JITM cache added in #46677
* Remove the ability for `:sidebar_notice` JITMs to be set differently per section.  Instead hardcode the messagePath to be the same, so pages like "Posts", "Pages", "Stats", etc all pull up the same JITMs from the left side.
* Remove the `messagePathSuffix` functionality from JITM.  It will only expect a basic `messagePath`.

#### Testing instructions

* Use calypso, both with `?flags=nav-unification` on and off

* The pop-up notices should continue to work correctly and not momentarily disappear and reappear as you navigate through the site
* The `<SiteNotice>` component shouldn't re-render excessively

### CODE REVIEW

**Comment by mreishus** [comment]
Need to examine/consider the comments on #46713 as well

**Comment by mreishus** [comment]
* Includes changes in [#46824 JITM understands both messagePath and messagePathSuffix props]  
* Still needs performance testing
* Still needs write-up + description

File: client/my-sites/theme/main.jsx [hunk 1]
```diff
@@ -345,7 +345,7 @@ class ThemeSheet extends React.Component {
 		return (
 			<div className="theme__sheet-content">
 				{ config.isEnabled( 'jitms' ) && this.props.siteSlug && (
-					<AsyncLoad require="calypso/blocks/jitm" messagePathSuffix={ 'admin_notices' } />
+					<AsyncLoad require="calypso/blocks/jitm" messagePath={ 'calypso:theme:admin_notices' } />
```
Comment (jsnajdr) [inline client/my-sites/theme/main.jsx:348]: This could use the JSX syntax for string props: `messagePath="calypso:..."`

**Review by jsnajdr — APPROVED** [review]
Looks good :+1: I'd prefer it over #46824 as it's more complete.

**Comment by mreishus** [comment]
Just adding Dave so he can see the latest developments when he gets back.  Review is not needed right now

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@mreishus please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/936147). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Comment by getdave** [comment]
Wow! So this continued to get more complex. Thanks for handling things whilst I was away. Nice job!

**Comment by mreishus** [comment]
> Wow! So this continued to get more complex. Thanks for handling things whilst I was away. Nice job!

No prob, you may want to check out my demo recording if you have some extra time: paYJgx-169-p2

**Pull request merged**